packages/nestjs-client/src/credentials/credential.service.ts

```diff
@@ -0,0 +1,337 @@
+            order: { createdTs: 'DESC' },
+            lock: { mode: 'pessimistic_write' },
+          })
+
+          if (!lastCred)
+            throw new Error(
+              'No valid registry definition found. Please restart the service and ensure the module is imported correctly',
+            )
+          lastCred.revocationRegistryIndex = -1
+        }
+
+        const newCredential = await transaction.save(CredentialEntity, {
+          connectionId,
+          credentialDefinitionId,
+          revocationDefinitionId: lastCred.revocationDefinitionId,
+          revocationRegistryIndex: lastCred.revocationRegistryIndex + 1,
+          ...(refIdHash ? { refIdHash } : {}),
+          maximumCredentialNumber: this.maximumCredentialNumber,
+        })
+        return {
+          revocationRegistryDefinitionId: newCredential.revocationDefinitionId,
+          revocationRegistryIndex: newCredential.revocationRegistryIndex,
+        }
+      },
+    )
+
+    await this.apiClient.messages.send(
+      new CredentialIssuanceMessage({
+        connectionId,
+        credentialDefinitionId,
+        revocationRegistryDefinitionId,
+        revocationRegistryIndex,
+        claims: claims,
+      }),
+    )
+    if (revocationRegistryIndex === this.maximumCredentialNumber - 1) {
+      const revRegistry = await this.saveCredentialType(credentialDefinitionId, revocationSupported)
+      this.logger.log(`Revocation registry successfully created with ID ${revRegistry}`)
+    }
+    this.logger.debug('sendCredential with claims: ' + JSON.stringify(claims))
+  }
+
+  /**
+   * Accepts a credential by associating it with the provided thread ID.
+   * @param connectionId - The connection ID associated with the credential.
+   * @param threadId - The thread ID to link with the credential.
+   * @throws Error if no credential is found with the specified connection ID.
+   */
+  async accept(connectionId: string, threadId: string): Promise<void> {
+    const cred = await this.credentialRepository.findOne({
+      where: {
+        connectionId,
+        revoked: false,
+      },
+      order: { createdTs: 'DESC' },
+    })
+    if (!cred) throw new Error(`Credential not found with connectionId: ${connectionId}`)
+
+    cred.threadId = threadId
+    await this.credentialRepository.save(cred)
+  }
+
+  /**
+   * Rejected a credential by associating it with the provided thread ID.
+   * @param connectionId - The connection ID associated with the credential.
+   * @param threadId - The thread ID to link with the credential.
+   * @throws Error if no credential is found with the specified connection ID.
+   */
+  async reject(connectionId: string, threadId: string): Promise<void> {
+    const cred = await this.credentialRepository.findOne({
+      where: {
+        connectionId,
+        revoked: false,
+      },
+      order: { createdTs: 'DESC' },
+    })
+    if (!cred) throw new Error(`Credencial with connectionId ${connectionId} not found.`)
```

Comment:
```suggestion
    if (!cred) throw new Error(`Credential with connectionId ${connectionId} not found.`)
```

As explained before, threadId should be used instead of connectionId.

The connectionId is only needed to determine where to send the notification in case of a revocation.